src/pages/profile/profile.js

```diff
@@ -1,127 +1,66 @@
 import "./profile.styles.css";
-import { Header, Footer } from "components";
-import { useState } from "react";
-import { Link, NavLink } from "react-router-dom";
-
-function getUserInfo() {
-  return JSON.parse(localStorage.getItem("userInfo"));
-}
+import { Loader } from "components";
+import { useAuthContext } from "context";
+import { useState, useEffect } from "react";
 
 function Profile() {
-  const [isNavFullLength, setIsNavFullLength] = useState(false);
-  const userInfo = getUserInfo();
+  const { authState } = useAuthContext();
+  const [userDetails, setUserDetails] = useState(false);
+  const [loader, setLoader] = useState("flex");
 
   //@TODO: This says I dont understand "this"
   function getItem(item) {
-    return userInfo[item] ?? "- not added -";
+    return userDetails[item] ?? "- not added -";
   }
 
+  function getuserInfo() {
+    const userInfo = JSON.parse(localStorage.getItem("userInfo"));
+    return userInfo;
+  }
+
+  useEffect(() => {
+    const userInfo = getuserInfo();
+    setLoader("none");
+    setUserDetails(userInfo);
+  }, []);
+
   return (
-    <div className="profile-page-container">
-      <Header />
-      <main className="profile-main">
-        <div className="profile-meta-info">
-          <h2>Account</h2>
-          <p>
-            {getItem("firstName")} {getItem("lastName")}
-          </p>
+    <section className="profile-section common-section">
+      {userDetails && <h3 className="section-header">Profile Details</h3>}
+      {userDetails && (
+        <div className="section-body">
+          <div className="profile-info">
+            <p className="info-header">Email ID</p>{" "}
+            <p className="info-value">{getItem("email")}</p>
+          </div>
```

Comment:
you may create this as a component, as these codes are being used multiple times

Reply:
Applied